pkg/memtier/orderedMap.go

```diff
@@ -0,0 +1,113 @@
+package memtier
+
+import (
+	rbt "github.com/emirpasic/gods/trees/redblacktree"
+)
+
+// OrderedMap represents an ordered map data structure.
+// It combines the functionalities of a Red-Black Tree and a key-value map.
+// The keys are uint64 values, and the values are of any type specified by the user.
+type OrderedMap[V any] struct {
+	rbt.Tree // Tree is the underlying Red-Black Tree used for maintaining the order of keys.
+}
+
+func compareUint64(a, b interface{}) int {
+	if a.(uint64) < b.(uint64) {
+		return -1
+	} else if a.(uint64) > b.(uint64) {
+		return 1
+	}
+	return 0
+}
+
+// NewOrderedMap creates and returns a new instance of OrderedMap with the specified value type.
+func NewOrderedMap[V any]() *OrderedMap[V] {
+	return &OrderedMap[V]{
+		Tree: rbt.Tree{
+			// Set the Comparator function to compareUint64 for sorting keys.
+			Comparator: compareUint64,
+		},
+	}
+}
+
+// Get returns the value for a key.
+// If the key does not exist, returns nil and false.
+func (m *OrderedMap[V]) Get(key uint64) (value V, ok bool) {
+	v, ok := m.Ceiling(key)
```

Comment:
When OrderedMap contains two heatranges (startaddr-endaddr): 2-3 and 5-6, they are stored to keys 2 and 5. When user calls orderedmap.Get(3), does m.Ceiling(3) here mean that range 5-6 will be returned?

Reply:
Range 2-3 will be returned.